conda_build/variants.py

```diff
@@ -745,15 +745,31 @@ def find_used_variables_in_text(variant, recipe_text, selectors_only=False):
         v_req_regex = "[-_]".join(map(re.escape, v.split("_")))
         variant_regex = rf"\{{\s*(?:pin_[a-z]+\(\s*?['\"])?{v_regex}[^'\"]*?\}}\}}"
         selector_regex = rf"^[^#\[]*?\#?\s\[[^\]]*?(?<![_\w\d]){v_regex}[=\s<>!\]]"
+        # NOTE: why use a regex instead of the jinja2 parser/AST?
+        # One can ask the jinj2 parser for undefined variables, but conda-build moves whole
+        # blocks of text around when searching for variables and applies selectors to the text.
+        # So the text that reaches this function is not necessarily valid jinja2 syntax. :/
```

Comment:
@jaimergp @wolfv @h-vetinari I have added a comment explaining why we have to use a regex here. Let's merge this one instead of letting the perfect be the enemy of the good.

Reply:
yeah, I think at this point it's impossible to fix this properly, sadly. LGTM!